pkg/config/settings.go

```diff
@@ -85,98 +86,132 @@ func (s *Settings) GetRequiredVersion(name string) string {
+			// however as .NET has a different ruleset for this,
+			switch cmd.Name {
+			case "dotnet":
 
-		case "nx":
+				// put together a path for the global.json file
+				globalData := filepath.Join(tmpPath, "global.json")
 
-			versionCmd = "node"
-			versionArgs = "--version"
-			re = *regexp.MustCompile(`"v(?P<version>.*)"`)
+				// determine if the file exists, if not see if there is a constraint set in the
+				// framework project settings
+				if !util.Exists(globalData) {
+					if cmd.Version != "" {
+						globalData = cmd.Version
+					}
+				}
 
-		default:
-			versionCmd = ""
-		}
+				// set the global value
+				version.DotNetGlobal(globalData)
 
-		// execute the command if one has been set
-		if versionCmd == "" {
-			continue
-		}
+				met, err = version.DotNet()
 
-		result, err := config.ExecuteCommand(path, logger, versionCmd, versionArgs, false, true)
+				// version_segments, err := util.VersionSegments(re, matches)
+				if err != nil {
+					logger.Errorf(err.Error())
+				}
 
-		// check for errors
-		if err != nil {
-			logger.Errorf("Issue running command: %s", err.Error())
-			continue
-		}
+				if err != nil {
+					resultErrors = append(resultErrors, err.Error())
+				}
 
-		// get the version from the result so it can be tested using semver
-		matches := re.FindStringSubmatch(result)
-		idx := re.SubexpIndex(("version"))
-		versionFound := matches[idx]
+			default:
 
-		logger.Debugf("Tool version found: %s", versionFound)
+				// for all other commands this is a semantic version check
+				// so use the "version" named group to compare against
 
-		// get the constraint that should be used to check for
-		// if a specific version has been specified modify this constraint
-		constraint := cmd.Version
-		if specificVersion != "" {
-			constraint = fmt.Sprintf("= %s", specificVersion)
-		}
+				// get the comparator to use and override the default if
+				// one has been provided
+
+				// If a version constraint has been specified, update the version object
+				// to use it
+				if cmd.Version != "" {
+					version.SetSemverConstraint(cmd.Version)
+				}
+
+				logger.Debugf("Version of '%s' found: %s", fCmd.Name, versionFound)
+
+				met, err = version.Semver()
+				if err != nil {
+					resultErrors = append(resultErrors, err.Error())
+				}
 
-		met := s.CompareVersion(constraint, versionFound, logger)
+			}
 
-		// if not matched then create a command object and set in the array
-		if !met {
-			incorrect = append(incorrect, models.Command{
-				Binary:          cmd.Name,
-				VersionFound:    versionFound,
-				VersionRequired: constraint,
-			})
+			// if not matched then create a command object and set in the array
+			if !met {
+				incorrect = append(incorrect, models.Command{
+					Binary:          cmd.Name,
+					VersionFound:    versionFound,
+					VersionRequired: constraint,
+				})
+			}
 		}
+	}
 
+	// if there any errors, log them
+	if len(resultErrors) > 0 {
```

Comment:
Do you need the if here? Won't it just not go into the loop if there's no errors?

Reply:
As the CLI can check the version of multiple tools, as it can work with multiple projects, it stores any errors up in a slice.
Then it checks to see if this slice has any items, if it does then it generates errors for each entry.
This was done so that it checks all the tools and outputs everything, rather than outputting an error and then stopping which can be tedious.

Reply:
But you range over it below? The for is superfluous?

Reply:
So I need to check that there are errors so we can exit out after ranging over all of the errors and outputting them as errors.